_posts/2019-10-16-deployment-updates.md

```diff
@@ -0,0 +1,214 @@
+or Kubernetes cluster.  In this the [Dask-Gateway](https://gateway.dask.org)
+project may be useful.
+It can launch and manage Dask jobs,
+and provide a proxy connection to these jobs if necessary.
+It is typically deployed with elevated permissions but managed directly by IT,
+giving them a point of greater control.
+
+<img src="https://gateway.dask.org/_images/architecture.svg" width="50%">
+
+
+GPUs and Dask-CUDA
+------------------
+
+While using Dask with multi-GPU deployments the [NVIDIA
+Rapids](https://rapids.ai) has needed the ability to specify increasingly
+complex setups of Dask workers.  They recommend the following deployment
+strategy:
+
+1.  One Dask-worker per GPU on a machine
+2.  Specify the `CUDA_VISIBLE_DEVICES` environment variable to pin that worker
+    to that GPU
+3.  If your machine has multiple network interfaces then choose the network interface closest to that GPU
+4.  If your machine has multiple CPUs then set thread affinities to use the closest CPU
+5.  ... and more
+
+For this reason we wanted to specify these configurations in code, like the
+following:
+
+```python
+specification = {
+    "worker-0": {
+        "cls": dask.distributed.Nanny,
+        "options": {"nthreads": 1, "env": {"CUDA_VISIBLE_DEVICES": "0,1,2,3"}, interface="ib0"},
+    },
+    "worker-1": {
+        "cls": dask.distributed.Nanny,
+        "options": {"nthreads": 1, "env": {"CUDA_VISIBLE_DEVICES": "1,2,3,0"}, interface="ib0"},
+    },
+    "worker-2": {
+        "cls": dask.distributed.Nanny,
+        "options": {"nthreads": 1, "env": {"CUDA_VISIBLE_DEVICES": "2,3,0,1"}, interface="ib1"},
+    },
+    "worker-2": {
+        "cls": dask.distributed.Nanny,
+        "options": {"nthreads": 1, "env": {"CUDA_VISIBLE_DEVICES": "3,0,1,2"}, interface="ib1"},
+    },
+}
+```
+
+And the new SpecCluster class to deploy these workers:
+
+```python
+cluster = SpecCluster(workers=specification)
+```
+
+This class was generic enough that it ended up forming the base of the SSH,
+Jobqueue, and Kubernetes solutions as well.
+
+
+Standards and Conventions
+-------------------------
+
+The solutions above are built by different teams that work in different companies.
+This is great because those teams have hands-on experience with the
+cluster managers in the wild, but has historically been somewhat challenging to
+standardize user experience.  This is particularly challenging when we build
+other tools like IPython widgets or the Dask JupyterLab extension, which want
+to interoperate with all of the Dask deployment solutions.
+
+The recent rewrite of Dask-SSH, Dask-Jobqueue, Dask-Kubernetes, and the new
+Dask-Cloudprovider and Dask-CUDA libraries place them
+all under the same `dask.distributed.SpecCluster` superclass.  So we can expect a high degree of
+uniformity from them.  Additionally, all of the classes now inherit from the
+`dask.distributed.Cluster` class, which standardizes things like adaptivity,
```

Comment:
Note: dask-gateway and dask-yarn don't inherit from `dask.distributed.Cluster`, but do match the API.